packages/jest/src/utils/config/get-projects.ts

```diff
@@ -0,0 +1,80 @@
+import { readWorkspaceConfig } from '@nrwl/workspace';
+import { dirname, join } from 'path';
+
+interface TargetConfig {
+  executor: string;
+  options: {
+    jestConfig: string;
+  };
+  configurations?: Record<
+    string,
+    { executor?: string; options?: { jestConfig?: string } }
+  >;
+}
+
+interface ProjectConfig {
+  targets?: Record<string, TargetConfig>;
+}
+
+export interface WorkspaceConfig {
+  projects: Record<string, ProjectConfig>;
+}
+
+const JEST_RUNNER_TOKEN = '@nrwl/jest:jest';
+
+function removeDuplicates<T>(arr: T[]): T[] {
+  const asSet = new Set<T>(arr);
+  return Array.from(asSet);
+}
+
+function getJestConfigProjectPath(projectJestConfigPath: string): string {
+  return join('<rootDir>', dirname(projectJestConfigPath));
+}
+
+function jestPathsByTargetExecutor(targets: TargetConfig[]): string[] {
+  return targets
+    .filter((target) => target.executor === JEST_RUNNER_TOKEN)
+    .map((target) => getJestConfigProjectPath(target.options.jestConfig));
+}
+
+function jestPathsFromTargetConfigurations(target: TargetConfig): string[] {
+  if (!target.configurations) {
+    return [];
+  }
+  return Object.values(target.configurations)
+    .reduce((acc, config) => {
+      if (!config.executor || config.executor !== JEST_RUNNER_TOKEN) {
+        return acc;
+      }
+      acc.push(
+        (config.options && config.options.jestConfig) ||
+          target.options.jestConfig
+      );
+      return acc;
+    }, [] as string[])
+    .map(getJestConfigProjectPath);
+}
+
+export function getProjects() {
+  const ws = readWorkspaceConfig({
+    format: 'nx',
+  }) as WorkspaceConfig;
+  const allProjectsJestConfigPath = Object.values(ws.projects)
+    .filter((project): project is ProjectConfig => !!project.targets)
+    .reduce((projectsAcc, project) => {
+      const allProjectTargets = Object.values(project.targets);
+      const allProjectJestConfigByExecutor = jestPathsByTargetExecutor(
+        allProjectTargets
+      );
+      const allProjectJestConfigByTargetConfigurations = allProjectTargets.reduce(
+        (targetsAcc, target) =>
+          targetsAcc.concat(jestPathsFromTargetConfigurations(target)),
+        [] as string[]
+      );
+      return projectsAcc.concat(
```

Comment:
I would push directly into the accumulated values.

```
projectsAcc.push(...allProjectJestConfigByExecutor, ...allProjectJestConfigByTargetConfigurations);
```

It's more performant and since acc is local to reduce there wouldn't be any bad side-effects.